smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/ReactiveKafkaConsumer.java

```diff
@@ -173,6 +175,20 @@ public void setRebalanceListener() {
                 .onItem().transformToMulti(v -> stream);
     }
 
+    public Multi<ConsumerRecords<K, V>> subscribeBatch(Set<String> topics) {
+        return runOnPollingThread(c -> {
+            c.subscribe(topics, rebalanceListener);
+        })
+                .onItem().transformToMulti(v -> batchStream);
+    }
+
+    public Multi<ConsumerRecords<K, V>> subscribeBatch(Pattern topics) {
+        return runOnPollingThread(c -> {
+            c.subscribe(topics, rebalanceListener);
+        })
+                .onItem().transformToMulti(v -> batchStream);
+    }
+
```

Comment:
This got me thinking: we don't do anything to prevent multiple subscriptions, do we? (Or multiple subscriptions/batchSubscriptions, that's the same thing.) If the user calls `subscribe` (or `subscribeBatch`) multiple times, they always get the same stream back. Which sounds weird.

Reply:
Now that I spent more time on `KafkaRecordStream`, the fact that it returns the same stream doesn't surprise me. 
I think we can turn `subscribe` methods into package private as they are intended to be called only by the owning `KafkaSource`.